libutils/json.h

```diff
@@ -354,13 +354,19 @@ JsonElement *JsonArrayGet(const JsonElement *array, size_t index);
   */
 bool JsonArrayContainsOnlyPrimitives(JsonElement *array);
 
+typedef JsonElement *JsonLookup(void *ctx, const char **data);
+
 /**
   @brief Parse a string to create a JsonElement
+  @param ctx [in] Evaluation context for variable lookups
   @param data [in] Pointer to the string to parse
   @param json_out Resulting JSON object
   @returns See JsonParseError and JsonParseErrorToString
+
+  The ctx type is void so we don't have to include eval_context.h from
+  libpromises into libutil
   */
-JsonParseError JsonParse(const char **data, JsonElement **json_out);
+JsonParseError JsonParse(void *ctx, JsonLookup *lookup, const char **data, JsonElement **json_out);
```

Comment:
Can you please leave the `JsonParse` function as is, and make a new function `JsonParseWithLookup`? The former can call the latter with null arguments. Then we don't have to change it everywhere, and the interface stays clean.

Also, slightly more cosmetic, but I think `ctx` should be renamed to `map`, since in principle it has no relation to `ctx`, it can be anything. Same thing for the `JsonLookup` function above. The comment about `ctx` being used as the map can then be moved to libpromises, at the place where it is actually passed in, which is where this information belongs. The information about the `map` argument can simply be something like `map which the JsonLookup function will use to look up references`, or something like that.

In general there shouldn't be any references to ctx in libutils, neither comments nor otherwise, since it is supposed to be independent from it.